src/libsystemd/sd-login/sd-login.c

```diff
@@ -134,6 +135,206 @@ _public_ int sd_pid_get_cgroup(pid_t pid, char **cgroup) {
         return 0;
 }
 
+_public_ int sd_pidfd_get_session(int pidfd, char **session) {
+        pid_t pid_early, pid_late;
+        int r;
+
+        assert_return(pidfd >= 0, -EBADF);
+        assert_return(session, -EINVAL);
+
+        r = pidfd_get_pid(pidfd, &pid_early);
+        if (r < 0)
+                return r;
+
+        r = cg_pid_get_session(pid_early, session);
+        if (r < 0)
+                return IN_SET(r, -ENXIO, -ENOMEDIUM) ? -ENODATA : r;
```

Comment:
maybe just call sd_pid_get_session() here, so that error mangling is only done at one place